src/gdext/surface/properties.nim

```diff
@@ -252,63 +342,150 @@ template `@export_exp_easing`*[T: SomeUserClass; S: SomeProperty](
+    gdexport_range(iden, min, max, step= defaultUnit[S](), extra, alias)
+
+template gdexport_range*[T: SomeUserClass; S: SomeNumeric](
     name;
     getter: proc(self: T): S;
     setter: proc(self: T; value: S);
     min, max, step: S; extra: varargs[RangeArgument]): untyped =
   register_property(typedesc T, name, typedesc S, getter, setter,
     hint= propertyHintRange,
-    hint_string= @[$min, $max, $step].concat(@extra.mapIt($it)).join(","))
-
-template `@export_storage`*[T: SomeUserClass; S: SomeProperty](
+    hint_string= rangeHintString(min, max, step, extra))
+macro gdexport_range*[S: SomeNumeric](
+  iden: typedesc[S],
+  min, max, step: S,
+  extra: varargs[RangeArgument], 
+  alias: static[alias] = noAlias) = 
+    quote do:
+      # compiler bug mitigation https://github.com/nim-lang/Nim/issues/11769
+      const hintString = when `extra`.len > 0: 
+        rangeHintString(`min`, `max`, `step`, `extra`)
+      else:
+        rangeHintString(`min`, `max`, `step`)
+
+      register_property_iden(`iden`, `alias`, hint= propertyHintRange, hint_string= hintString)
+
+template gdexport_storage*[T: SomeUserClass; S: SomeProperty](
     name;
     getter: proc(self: T): S;
     setter: proc(self: T; value: S)): untyped =
   register_property(typedesc T, name, typedesc S, getter, setter,
     usage= {propertyUsageStorage})
+macro gdexport_storage*(iden: SomeProperty, alias: static[alias] = noAlias) =
+  quote do:
+    register_property_iden(`iden`, `alias`, usage= {propertyUsageStorage})
+
+
+
+macro processExports*(T: typed): untyped =
+  let innerType = T.getType[1][1]
+
+  let objType = innerType.getImpl
+  let objTypeInner = objType[0]
+  var objTypeStr = $objTypeInner.toStrLit
+  let index = objTypeStr.find(":")
+  objTypeStr = objTypeStr[0..<index]
+
+  let fields = objType[2][2]
+
+  let module = objType[0].owner
+  let classIdent = newIdentNode(objTypeStr)
+
+  var stmts = newStmtList()
+  
+  for field in fields:
+    if field[0].kind != nnkPragmaExpr:
+      continue
+
+    var fieldName = $field[0][0][1]
+    let fieldIdent = newIdentNode(fieldName)
+    let dotExpr = newDotExpr(newDotExpr(module, classIdent), fieldIdent)
+    
+    let pragma = field[0][1]
+    let pragmaKind = pragma[0].kind
+    let pragmaIdent = case pragmaKind:
+      of nnkSym:
+        $pragma[0]
+      of nnkExprColonExpr, nnkCall:
+        $pragma[0][0]
+      else:
+        error("Unexpected pragma configuration")
+    
+    let pragmaParams = pragma[0]
+ 
+    var args = if pragmaParams.len > 0:
+      var sq = @[dotExpr] 
+      for param in pragmaParams[1..^1]: 
+        # Ignore the case where compiler inserts concept type as a call param 
+        if param.kind == nnkSym: 
+          let impl = bindSym($param).getImpl
+          if impl.kind == nnkTypeDef:
+            continue
+        
+        sq.add param
+      sq
+    else:
+      @[dotExpr]
+
+    # echo args.repr
+    stmts.add bindSym(pragmaIdent).newCall args
+      
+  result = stmts
```

Comment:
This macro seems a bit too complicated.
There is a set of functions in utils/macros that simplify getting information from objects. If something is missing, please add it there.